src/controllers/routers/cases.js

```diff
@@ -74,14 +73,194 @@ function sanitizedCase(json) {
+    } else {
+      // to make include_entities in the [ ] format for .withGraphFetched, and remove in between spaces
+      query.include_entities = `[${queryArray.replace(/\s/g, '')}]`;
+    }
+  }
+  // to make status uppercase
+  if (json.status) {
+    query.status = json.status.toUpperCase();
+  }
+  // to retrieve sort field and sort order
+  if (json.sort) {
+    const array = json.sort.split(':');
+    [query.sort_field, query.sort_order] = [array[0].trim(), array[1].trim()];
+  }
+  // to check if applied_on is in YYYY-MM-DD format
+  if (
+    json.applied_on &&
+    /([12]\d{3}-(0[1-9]|1[0-2])-(0[1-9]|[12]\d|3[01]))/.test(
+      json.applied_on
+    ) === true
+  ) {
+    const year = json.applied_on.slice(0, 4);
+    const month = json.applied_on.slice(5, 7);
+    const day = json.applied_on.slice(8, 10);
+    query.applied_on = `${year}-${month}-${day}`;
+  }
+  return query;
+}
+
+/**
+ * Set default values based on query params.
+ * Call before an insert or an update.
+ * @param {Object} json - Query params without default values
+ * @return {Object} query - Query params with default values
+ */
+function setDefault(json) {
+  const query = json;
+  // if with_paging is not available or false, set page and per_page to null
+  if (!json.with_paging || json.with_paging !== 'true') {
+    query.page = null;
+    query.per_page = null;
+  }
+  // if with_paging is true
+  else {
+    // if page is defined in query param, otherwise default to 1
+    if (json.page !== undefined || json.page !== null) {
+      query.page = parseInt(json.page, 10);
+    } else {
+      query.page = 1;
+    }
+    // if per_page is defined in query param, otherwise default to 10
+    if (json.per_page !== undefined || json.per_page !== null) {
+      query.per_page = parseInt(json.per_page, 10);
+    } else {
+      query.per_page = 10;
+    }
+  }
+  // if status is not present, set default to 'ALL'
+  if (!json.status) {
+    query.status = 'ALL';
+  }
+  // if sort is not present, set default to arrange id in asc
+  if (!json.sort) {
+    query.sort_field = 'id';
+    query.sort_order = 'asc';
+  }
+  return query;
+}
+
 /**
  * Retrieve all cases
  * @param {Request} req
  * @param {Response} res
  */
-const getAll = async (req, res) => {
-  const cases = await Case.query().select();
-  return res.status(200).json({ cases });
+const getAll = async (req, res, next) => {
+  const parsedQueries = sanitizedQuery(req.query);
+
+  // to set the default values for with_paging, page, per_page and status
+  setDefault(parsedQueries);
+
+  const currentPage = parsedQueries.page;
+  const limit = parsedQueries.per_page;
+  // if offset is undefined, set it to 0
+  const offset = limit * currentPage - limit ? limit * currentPage - limit : 0;
+
+  // to retrieve all cases to find the total number of cases for the response object
+  const allCases = await Case.query();
+  try {
+    const results = await Case.query()
```

Comment:
I get errors when I apply some of the filters, e.g., beneficiary name, referee name, etc

Reply:
include an error message to tell user that beneficiary/name etc is not found in include_entities